### ISSUE TITLE
Icons are rendered twice

### ISSUE THREAD

**Issue by netcitylife** [jmzagorski/aurelia-fontawesome#16]
**I'm submitting a bug report**

* **Library Version:**
0.2.1

* **Browser:**
all

**Current behavior:**
iconChanged() hook calls attached() hook, so attahed() is called twice of first render (first when icon's value changed from undefined to what was assined from outside, and then when component is actually attached to DOM).

**What is the expected behavior?**
Icon is rendered once and rerendered only when it actually changes.

**What is the motivation / use case for changing the behavior?**
iconChanged() should not call attached() directly (this is generally a bad practice to call one public hook from another because of such cases), instead some private method should be extracted for that purpose and then in might be called from both attached() and iconChanged()

**Comment by jmzagorski** [comment]
Ahh that is right, I removed `bind`  when I removed all the ViewSlot logic, which prevented the iconChanged from firing during the lifecycle. Good catch again!

**Comment by jmzagorski** [comment]
Probably can just add an `_isAttached` private field for these change handlers and not fire the logic in `iconChanged` or `propertyChanged` when `_isAttached` is false

**Comment by netcitylife** [comment]
First - it's best to handle all renderings (including initial rendering) in `iconChanged`, not in attached(), there's no need to wait for DOM attachment for performance reasons.
Second - what does `propertyChanged` stand for? doesit fire on any component property change? I cannot find such global hook anywhere in the docs... does it really work? And if it does - doesn't it fire for icon property change also?

**Comment by jmzagorski** [comment]
`propertyChanged` signature is `propertChanged(propertyName, newValue, oldValue)` and yes it fires for every bindable property.  I realized this yesterday and removed `iconChanged` from the code where I am fixing this. I made the incorrect assumption, without looking at Aurelia's source code, that it would not fire `propertyChanged` for the icon property if I handled that logic in the `iconChanged` method.

I also realized that we want to handle the initialization code in `bind` and remove `attached`. We do not want to use `propertyChanged` because that will fire for every property when those values are binding to the custom element.  `bind` will prevent the `propertyChange` handler from firing during initialization and after initialization `propertyChange` will fire as expected.
